extensions/markdown-language-features/preview-src/index.ts

```diff
@@ -146,23 +147,31 @@ async function copyImage(image: HTMLImageElement, retries = 5) {
 	}
 
 	try {
-		await navigator.clipboard.write([new ClipboardItem({
-			'image/png': new Promise((resolve) => {
+		const imageSrc = image.getAttribute('data-src') || 'src-not-found';
+
+		if (isURL(imageSrc)) {
+			await navigator.clipboard.writeText(imageSrc);
+		} else {
+			const blobImage: Blob = await new Promise((resolve) => {
 				const canvas = document.createElement('canvas');
+
 				if (canvas !== null) {
 					canvas.width = image.naturalWidth;
 					canvas.height = image.naturalHeight;
 					const context = canvas.getContext('2d');
 					context?.drawImage(image, 0, 0);
 				}
+
 				canvas.toBlob((blob) => {
 					if (blob) {
 						resolve(blob);
 					}
 					canvas.remove();
 				}, 'image/png');
-			})
-		})]);
+			});
+
+			await navigator.clipboard.write([new ClipboardItem({ 'image/png': blobImage })]);
```

Comment:
I'd revert this too. You want to call `write` early so that browsers know there is pending image data. Now it will instead wait for the blob to be created and only then write the clipboard. It's possible that clipboard changes between those steps